subsys/net/ip/ipv4_autoconf.c

```diff
@@ -0,0 +1,288 @@
+static void send_announcement(struct net_if *iface);
+
+static inline struct net_buf *prepare_arp(struct net_if *iface)
+{
+	struct net_buf *buf, *frag;
+	struct net_arp_hdr *hdr;
+	struct net_eth_hdr *eth;
+
+	buf = net_nbuf_get_reserve_tx(0, K_FOREVER);
+	if (!buf) {
+		goto fail;
+	}
+
+	frag = net_nbuf_get_reserve_data(sizeof(struct net_eth_hdr), K_FOREVER);
+	if (!frag) {
+		goto fail;
+	}
+
+	net_buf_frag_add(buf, frag);
+	net_nbuf_set_iface(buf, iface);
+	net_nbuf_set_family(buf, AF_INET);
+	net_nbuf_set_ll_reserve(buf, sizeof(struct net_eth_hdr));
+
+	eth = NET_ETH_BUF(buf);
+	hdr = NET_ARP_BUF(buf);
+	eth->type = htons(NET_ETH_PTYPE_ARP);
+
+	/* ethernet header */
+	memcpy(&eth->src.addr, net_if_get_link_addr(iface)->addr,
+	       sizeof(struct net_eth_addr));
+	memset(&eth->dst.addr, 0xFF, sizeof(struct net_eth_addr));
+
+	hdr->hwtype = htons(NET_ARP_HTYPE_ETH);
+	hdr->protocol = htons(NET_ETH_PTYPE_IP);
+	hdr->hwlen = sizeof(struct net_eth_addr);
+	hdr->protolen = sizeof(struct in_addr);
+	hdr->opcode = htons(NET_ARP_REQUEST);
+
+	memcpy(&hdr->src_hwaddr.addr, &eth->src.addr,
+	       sizeof(struct net_eth_addr));
+	memset(&hdr->dst_hwaddr.addr, 0x00, sizeof(struct net_eth_addr));
+
+	net_ipaddr_copy(&hdr->dst_ipaddr, &iface->ipv4_autoconf.current_ip);
+	net_ipaddr_copy(&hdr->src_ipaddr, &iface->ipv4_autoconf.requested_ip);
+
+	net_buf_add(frag, sizeof(struct net_arp_hdr));
+
+	return buf;
+
+fail:
+	net_nbuf_unref(buf);
+	return NULL;
+}
+
+enum net_verdict net_ipv4_autoconf_input(struct net_if *iface,
+					 struct net_buf *buf)
+{
+	struct net_arp_hdr *arp_hdr;
+
+	if (net_buf_frags_len(buf) <
+	    (sizeof(struct net_arp_hdr) - net_nbuf_ll_reserve(buf))) {
+		NET_DBG("Invalid ARP header (len %zu, min %zu bytes)",
+			net_buf_frags_len(buf),
+			sizeof(struct net_arp_hdr) - net_nbuf_ll_reserve(buf));
+		return NET_DROP;
+	}
+
+	arp_hdr = NET_ARP_BUF(buf);
+
+	if (!net_ipv4_addr_cmp(&arp_hdr->dst_ipaddr,
+			       &iface->ipv4_autoconf.requested_ip)) {
+		/* No conflict */
+		return NET_DROP;
+	}
+
+#if NET_DEBUG > 0
+	do {
+		NET_DBG("conflict detected from %s [%s] for %s", out,
+			net_sprint_ll_addr((uint8_t *)&arp_hdr->src_hwaddr,
+					   arp_hdr->hwlen),
+			net_sprint_ipv4_addr(&arp_hdr->dst_ipaddr));
+	} while (0);
+#endif
+
+	iface->ipv4_autoconf.conflict_cnt++;
+
+	switch (iface->ipv4_autoconf.state) {
+	case NET_IPV4_AUTOCONF_PROBE:
+		/* restart probing with renewed IP */
+		net_ipv4_autoconf_start(iface);
+		break;
+	case NET_IPV4_AUTOCONF_ANNOUNCE:
+	case NET_IPV4_AUTOCONF_CONFIGURED:
+		if (iface->ipv4_autoconf.conflict_cnt == 1) {
+			/* defend IP */
+			send_announcement(iface);
+		} else {
+			/* unset host ip */
+			if (!net_if_ipv4_addr_rm(
+				iface, &iface->ipv4_autoconf.requested_ip)) {
+				NET_DBG("Failed to remove addr from iface");
+			}
+			/* restart probing after second conflict */
+			net_ipv4_autoconf_start(iface);
+		}
+		break;
+	default:
+		break;
+	}
+	return NET_DROP;
```

Comment:
break a line after a } (unless it's another } etc...) apply this rule everywhere